Add Xamarin implementation

jschmid opened the pull request:
For a project using Xamarin, I had to integrate libsodium. I could not find readily available packages. 

There are minimal changes needed to make it work:

* Update the way `DllImport` is called, because of the way Xamarin.iOS packages dlls
* Use `msbuild` instead of `dotnet`, because dotnet does not support Xamarin

For now, the binary files for Android and iOS still have to be included manually to the project, because the `libsodium` Nuget itself does not contain them, but at least the "C# wrapper" part can be automated. 

Instead of publishing my own nuget under a different name, I though it would be nicer to submit a PR first to see what you thought about it.

The PR has to be improved a bit (squashing commits, updating the readme), but I'll wait for your feedback before spending more time on it.

What do you think ?

jschmid commented:
Looks like codecov is broken on master.

jschmid commented:
FYI I published a custom version [here](https://github.com/liip/Sodium.Xamarin). My goal would be to deprecate it if my PR can be merged here.

tabrath commented on src/Sodium.Core/Sodium.Core.csproj:
```diff
@@ -1,15 +1,15 @@
     <Description>libsodium for .net core</Description>
     <Copyright>Copyright © tabrath 2017</Copyright>
-    <AssemblyTitle>Sodium.Core</AssemblyTitle>
+    <AssemblyTitle>Sodium.Xamarin</AssemblyTitle>
```
I assume you just forgot about this change, right?

tabrath commented on src/Sodium.Core/Sodium.Core.csproj:
```diff
@@ -1,15 +1,15 @@
-    <AssemblyName>Sodium.Core</AssemblyName>
-    <PackageId>Sodium.Core</PackageId>
+    <AssemblyName>Sodium.Xamarin</AssemblyName>
+    <PackageId>Sodium.Xamarin</PackageId>
```
Same here.

jschmid commented:
Thanks @tabrath. I wanted to make sure that you would accept the PR before doing a final pass on all my changes. I now updated and rebased my work.

You can see that I added a separate document explaining how to use the library with Xamarin, because you still need to compile _libsodium_ yourself, as the NuGet does not provide it for Xamarin.

jschmid commented:
**I noticed a bug. Please do not merge yet!**

jschmid commented:
@tabrath sorry for the back and forth. There was still a bug with the NuGet package. I fixed it, and now the diff is even cleaner.

You can merge if it's all good on your side.
